coderd/httpapi/httpapi_test.go

```diff
@@ -155,3 +160,263 @@ func TestWebsocketCloseMsg(t *testing.T) {
 		assert.Equal(t, len(trunc), 123)
 	})
 }
+
+// Our WebSocket library accepts any arbitrary ResponseWriter at the type level,
+// but it must also implement http.Hijack
+type mockWsResponseWriter struct {
+	serverRecorder   *httptest.ResponseRecorder
+	serverConn       net.Conn
+	clientConn       net.Conn
+	serverReadWriter *bufio.ReadWriter
+}
+
+func (m mockWsResponseWriter) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+	return m.serverConn, m.serverReadWriter, nil
+}
+
+func (m mockWsResponseWriter) Flush() {
+	_ = m.serverReadWriter.Flush()
+}
+
+func (m mockWsResponseWriter) Header() http.Header {
+	return m.serverRecorder.Header()
+}
+
+func (m mockWsResponseWriter) Write(b []byte) (int, error) {
+	return m.serverReadWriter.Write(b)
+}
+
+func (m mockWsResponseWriter) WriteHeader(code int) {
+	m.serverRecorder.WriteHeader(code)
+}
+
+type mockWsWrite func(b []byte) (int, error)
+
+func (w mockWsWrite) Write(b []byte) (int, error) {
+	return w(b)
+}
+
+func TestOneWayWebSocket(t *testing.T) {
```

Comment:
Because we also touch the SSE Sender, could we include some basic coverage for that if there isn't already?